### PR TITLE
Add unit conversions for Duration, MassFraction and add additional Density Convert methods

### LINKED ISSUE #97: Add conversion for concentrations (density and mass fraction)
#### Description:

We need to add `Convert` methods for density (i.e. mg/l, μg/l) to SI units whereby g/l is the SI equivalent. See below for naming to be used:  
https://github.com/angularsen/UnitsNet/blob/master/Common/UnitDefinitions/Density.json

We need to add `Convert` methods for mass fraction (i.e. mg/kg, μg/kg): 
https://github.com/angularsen/UnitsNet/blob/master/Common/UnitDefinitions/MassFraction.json

There is already a folder for density conversions below, we need to make a new one for Mass Fractions:
https://github.com/BHoM/Localisation_Toolkit/tree/develop/Units_Engine/Convert/Density

### PR DESCRIPTION
**Pull request by Martian42**
### NOTE: Depends on 

 ### Issues addressed by this PR

 Closes #97 

 

 ### Test files

https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/Localisation_Toolkit/Units_Engine/%2398-Add%20relevant%20engine%20methods%20and%20enum%20associated%20with%20new%20AGS%20object?csf=1&web=1&e=gT9d9K

 ### Changelog

- Added `Convert` methods for specific `Density` methods such as `MicrogramPerLitre` and `MilligramPerLitre`;
- Added `Duration` `Convert` methods for `Millisecond`, `Minute` and `Hour`;
- Added `MassFraction` `Convert` methods for `NanogramPerKillogram`, `MicrogramPerKillogram`, `GramPerKilogram`, `CentigramPerKilogram`, `DecigramPerKilogram`;
- Added `eNums` for `MassFraction` and `Duration`.

 ### Additional comments

Mass fraction conversion methods added in this PR. In responds to issue #97 , density conversion is already available, no additional methods needed.

### CODE REVIEW

File: Units_oM/Enums/MassFraction.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.oM.Units
+{
+    public enum MassFractionUnit
+    {
+        Undefined = 0,
+        NanogramPerKilogram = 1,
```
Comment (FraserGreenroyd) [inline Units_oM/Enums/MassFraction.cs:34]: This should be in alphabetical order

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by peterjamesnugent** [comment]
@Martian42 can we add some time conversions please?

milliseconds, seconds, minutes, hours should suffice.

**Review by peterjamesnugent — CHANGES_REQUESTED** [review]
Happy that the `MassFraction` conversions are working correctly - as previous comment. Can we add the following:
- `Convert` methods for the Density units mentioned in the original issue (i.e. `ToMilligramPerLitre` and `FromMilligramPerLitre` etc.
- Add `Enums` and `Convert` methods for time, see https://github.com/angularsen/UnitsNet/blob/master/Common/UnitDefinitions/Duration.json

**Comment by Martian42** [comment]
> Happy that the `MassFraction` conversions are working correctly - as previous comment. Can we add the following:
> 
> * `Convert` methods for the Density units mentioned in the original issue (i.e. `ToMilligramPerLitre` and `FromMilligramPerLitre` etc.
> * Add `Enums` and `Convert` methods for time, see https://github.com/angularsen/UnitsNet/blob/master/Common/UnitDefinitions/Duration.json



> Happy that the `MassFraction` conversions are working correctly - as previous comment. Can we add the following:
> 
> * `Convert` methods for the Density units mentioned in the original issue (i.e. `ToMilligramPerLitre` and `FromMilligramPerLitre` etc.
> * Add `Enums` and `Convert` methods for time, see https://github.com/angularsen/UnitsNet/blob/master/Common/UnitDefinitions/Duration.json

Thanks for your review @peterjamesnugent. I'll proceed with adding time methods. Density however, I believe they are already available.
image
image

File: Units_oM/Enums/MassFraction.cs (hunk 1 above)
Comment (FraserGreenroyd) [reply Units_oM/Enums/MassFraction.cs]: Size order makes sense, I didn't realise they were originally so apologies 😄

Comment (Martian42) [reply Units_oM/Enums/MassFraction.cs]: @FraserGreenroyd @peterjamesnugent @al-fisher agree! I'll use size order for all enum on the next push.

File: Units_Engine/Convert/MassFraction/NanogramPerKilogram.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+namespace BH.Engine.Units
+{
+    public static partial class Convert
+    {
+        [Description("Convert SI units (kilogram per kilogram) into nanogram per kilogram")]
+        [Input("kilogramPerKilogram", "The number of kilograms per kilogram to convert", typeof(MassFraction))]
+        [Output("nanogramPerKilogram", "The number of nanograms per kilogram")]
+        public static double ToNanogramPerKilogram(this double kilogramsPerKilogram)
+        {
+            UN.QuantityValue qv = kilogramsPerKilogram;
```
Comment (peterjamesnugent) [inline Units_Engine/Convert/MassFraction/NanogramPerKilogram.cs:45]:
```suggestion
        [Description("Convert SI units (kilograms per kilogram) into nanograms per kilogram")]
        [Input("kilogramsPerKilogram", "The number of kilograms per kilogram to convert", typeof(MassFraction))]
        [Output("nanogramsPerKilogram", "The number of nanograms per kilogram")]
        public static double ToNanogramPerKilogram(this double kilogramsPerKilogram)
        {
            UN.QuantityValue qv = kilogramsPerKilogram;
```

File: Units_Engine/Convert/MassFraction/NanogramPerKilogram.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            return UN.UnitConverter.Convert(qv, MassFractionUnit.KilogramPerKilogram, MassFractionUnit.NanogramPerKilogram);
+        }
+
+        [Description("Convert nanogram per kilogram into SI units (kilogram per kilogram)")]
+        [Input("nanogramPerKilogram", "The number of nanograms per kilogram to convert")]
+        [Output("kilogramPerKilogram", "The number of kilogram per kilogram", typeof(MassFraction))]
+        public static double FromNanogramPerKilogram(this double nanogramsPerKilogram)
+        {
+            UN.QuantityValue qv = nanogramsPerKilogram;
```
Comment (peterjamesnugent) [inline Units_Engine/Convert/MassFraction/NanogramPerKilogram.cs:54]:
```suggestion
        [Description("Convert nanograms per kilogram into SI units (kilograms per kilogram)")]
        [Input("nanogramsPerKilogram", "The number of nanograms per kilogram to convert")]
        [Output("kilogramsPerKilogram", "The number of kilograms per kilogram", typeof(MassFraction))]
        public static double FromNanogramPerKilogram(this double nanogramsPerKilogram)
        {
            UN.QuantityValue qv = nanogramsPerKilogram;
```

**Review by peterjamesnugent — CHANGES_REQUESTED** [review]
Tested using script, mostly happy.

Mostly changes to descriptions.

**Comment by peterjamesnugent** [comment]
@BHoMBot check code-compliance
@BHoMBot check copyright-compliance
@BHoMBot check documentation-compliance
@BHoMBot check project-compliance

**Comment by peterjamesnugent** [comment]
@BHoMBot check code-compliance
@BHoMBot check copyright-compliance
@BHoMBot check documentation-compliance
@BHoMBot check project-compliance

**Comment by peterjamesnugent** [comment]
@BHoMBot fix documentation punctuation ref. 18843030124

**Comment by FraserGreenroyd** [comment]
@BHoMBot fix documentation punctuation ref. 18843030124

**Comment by FraserGreenroyd** [comment]
@BHoMBot check documentation-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check documentation-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning
@BHoMBot check installer

**Review by peterjamesnugent — APPROVED** [review]
Happy with changes made as they all relate to the descriptions and object names.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**